File: Examples/runtimes/java/Migration/PlaintextToAWSDBE/src/main/java/software/amazon/cryptography/examples/plaintext/CreateSimpleTable.java
// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0
package software.amazon.cryptography.examples.plaintext;

import software.amazon.awssdk.services.dynamodb.DynamoDbClient;
import software.amazon.awssdk.services.dynamodb.model.AttributeDefinition;
import software.amazon.awssdk.services.dynamodb.model.CreateTableRequest;
import software.amazon.awssdk.services.dynamodb.model.CreateTableResponse;
import software.amazon.awssdk.services.dynamodb.model.KeySchemaElement;
import software.amazon.awssdk.services.dynamodb.model.KeyType;
import software.amazon.awssdk.services.dynamodb.model.ScalarAttributeType;

/**
 * This class is referenced by the README.
 */
@SuppressWarnings("unused")
public class CreateSimpleTable {

  public static void Create(DynamoDbClient ddbClient, String ddbTableName) {
    CreateTableRequest request = CreateTableRequest
      .builder()
      .tableName(ddbTableName)
      .keySchema(
        KeySchemaElement
          .builder()
          .keyType(KeyType.HASH)
          .attributeName("partition_key")
          .build(),
        KeySchemaElement
          .builder()
          .keyType(KeyType.RANGE)
          .attributeName("sort_key")
          .build()
      )
      .attributeDefinitions(
        AttributeDefinition
          .builder()
          .attributeName("partition_key")
          .attributeType(ScalarAttributeType.S)
          .build(),
        AttributeDefinition
          .builder()
          .attributeName("sort_key")
          .attributeType(ScalarAttributeType.N)
          .build()
      )
      .build();
    CreateTableResponse response = ddbClient.createTable(request);
    if (!response.sdkHttpResponse().isSuccessful()) {
      throw new RuntimeException(
        String.format(
          "Create Table Failed. HTTP response: %s",
          response.sdkHttpResponse()
        )
      );
    }
  }
}

File: Examples/runtimes/go/migration/PlaintextToAWSDBE/plaintext/createsimpletable.go
package plaintext

import (
	"context"

	"github.com/aws/aws-database-encryption-sdk-dynamodb/releases/go/dynamodb-esdk/examples/utils"
	"github.com/aws/aws-sdk-go-v2/service/dynamodb"
	"github.com/aws/aws-sdk-go-v2/service/dynamodb/types"
)

// CreateSimpleTable creates a DynamoDB table with partition_key (S) and sort_key (N)
func CreateSimpleTable(ddbClient *dynamodb.Client, ddbTableName string) {
	input := &dynamodb.CreateTableInput{
		TableName: &ddbTableName,
		KeySchema: []types.KeySchemaElement{
			{
				AttributeName: &[]string{"partition_key"}[0],
				KeyType:       types.KeyTypeHash,
			},
			{
				AttributeName: &[]string{"sort_key"}[0],
				KeyType:       types.KeyTypeRange,
			},
		},
		AttributeDefinitions: []types.AttributeDefinition{
			{
				AttributeName: &[]string{"partition_key"}[0],
				AttributeType: types.ScalarAttributeTypeS,
			},
			{
				AttributeName: &[]string{"sort_key"}[0],
				AttributeType: types.ScalarAttributeTypeN,
			},
		},
		BillingMode: "PAY_PER_REQUEST",
	}

	_, err := ddbClient.CreateTable(context.TODO(), input)
	utils.HandleError(err)
}
